### PR TITLE
Create UBI minimal base image

### PR DESCRIPTION
**Pull request by grooverdan**
There was a requirement for a UBI base image.

Entrypoint and healthcheck.sh are the same as the previous ubuntu image.

### CODE REVIEW

**Review by tianon — COMMENTED** [review]
A few notes, as requested :innocent: :heart:

File: 10.11-ubi/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+	chmod ugo+rwx,o+t /run/mariadb && \
+	microdnf install -y MariaDB-backup-10.11.8 MariaDB-server-10.11.8 && \
+	# compatibility with DEB Galera packaging
+	ln -s /usr/lib64/galera-4/libgalera_smm.so /usr/lib/libgalera_smm.so && \
+	# compatibility with RPM Galera packaging
+	ln -s /usr/lib64/galera-4/libgalera_smm.so /usr/lib64/galera/libgalera_smm.so && \
+	microdnf clean all && \
+	rmdir /var/lib/mysql/mysql && \
+	chown -R mysql:mysql /var/lib/mysql /run/mariadb && \
+	mkdir  /licenses && ln -s /usr/share/doc/MariaDB-server-10.11.8/COPYING /licenses/GPL-2
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply 10.11-ubi/Dockerfile:48]: This is part of the HasLicenses test of OpenShift certification - https://access.redhat.com/documentation/en-us/red_hat_software_certification/2024/html/red_hat_openshift_software_certification_policy_guide/assembly-requirements-for-container-images_openshift-sw-cert-policy-introduction

File: 10.11-ubi/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+# procps, pv(epel) - missing dependencies of galera sst script
+# tzdata re-installed as only a fake version is part of the ubi-minimal base image.
+# FF8AD1344597106ECE813B918A3872BF3228467C is the fedora RPM key
+RUN curl https://pagure.io/fedora-web/websites/raw/master/f/sites/getfedora.org/static/keys/FF8AD1344597106ECE813B918A3872BF3228467C.txt --output /tmp/epelkey.txt && \
+	rpmkeys --import /tmp/epelkey.txt && \
+	rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
+	rm /tmp/epelkey.txt && \
+	rpmkeys --import https://supplychain.mariadb.com/MariaDB-Server-GPG-KEY && \
+	microdnf update -y && \
+	microdnf reinstall -y tzdata && \
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply 10.11-ubi/Dockerfile:36]: There's a comment above the step.

File: 10.11-ubi/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,60 @@
+FROM redhat/ubi9-minimal
+
+# user 999/ group 999, that we want to use for compatibility with the ubuntu image.
+RUN groupadd --gid 999 -r mysql && \
+	useradd -r -g mysql mysql --home-dir /var/lib/mysql --uid 999
+
+ARG TARGETARCH
+ENV GOSU_VERSION 1.17
+RUN ARCH=$TARGETARCH && if [ $TARGETARCH = ppc64le ]; then ARCH=ppc64el ; fi ; \
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply 10.11-ubi/Dockerfile:9]: Nice, thank you. small improvement (that I don't need) `s390x|riscv64) dpkgArch=$rpmArch  ;;`

File: 10.11-ubi/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+COPY docker.cnf /etc/my.cnf.d/
+
+COPY MariaDB.repo /etc/yum.repos.d/
+
+# missing pwgen(epel), jemalloc(epel) (as entrypoint/user extensions)
+# procps, pv(epel) - missing dependencies of galera sst script
+# tzdata re-installed as only a fake version is part of the ubi-minimal base image.
+# FF8AD1344597106ECE813B918A3872BF3228467C is the fedora RPM key
+RUN curl https://pagure.io/fedora-web/websites/raw/master/f/sites/getfedora.org/static/keys/FF8AD1344597106ECE813B918A3872BF3228467C.txt --output /tmp/epelkey.txt && \
+	rpmkeys --import /tmp/epelkey.txt && \
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply 10.11-ubi/Dockerfile:31]: Yes, is a GPG key. Added validation.

The rpm install uses the keys, unfortunately without the key, it doesn't actually error
```
+ rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
warning: /var/tmp/rpm-tmp.qZSgy5: Header V4 RSA/SHA256 Signature, key ID 3228467c: NOKEY
Retrieving https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm
```

reword to use `rpm -K` to validate the package matches the key. and validate the key itself against its fingerprint.

File: 10.11-ubi/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply 10.11-ubi/Dockerfile:32]: I prefer the `rpm` based one, despite its verboseness which emphasises the generalised embarrassment of a linux ecosystem where we can't have a uniform description of what hardware is actually called. :smiling_face_with_tear:

File: 10.11-ubi/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	rm /tmp/epelkey.txt && \
+	rpmkeys --import https://supplychain.mariadb.com/MariaDB-Server-GPG-KEY && \
```
[the remark this answers is not in the document]
Comment (grooverdan) [reply 10.11-ubi/Dockerfile:34]: done. same way.

[the remark this answers is not in the document]
Comment (grooverdan) [reply 10.11-ubi/Dockerfile]: I checked what happened when this wasn't imported and:
```

error: package MariaDB-backup-10.6.18-1.el9.x86_64 cannot be verified and repo mariadb is GPG enabled: /var/cache/yum/metadata/mariadb-9-x86_64/packages/MariaDB-backup-10.6.18-1.el9.x86_64.rpm could not be verified.
/var/cache/yum/metadata/mariadb-9-x86_64/packages/MariaDB-backup-10.6.18-1.el9.x86_64.rpm:  digest:  SIGNATURE:  NOT OK
```
So a direct match of the rpm to the GPG is done by microdnf.

**Pull request merged**